src/control/server/drpc_test.go

```diff
@@ -0,0 +1,111 @@
+//
+// (C) Copyright 2019 Intel Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+// The Government's rights to use, modify, reproduce, release, perform, display,
+// or disclose this software are subject to the terms of the Apache License as
+// provided in Contract No. 8F-30005.
+// Any reproduction of computer software, computer software documentation, or
+// portions thereof marked with this legend must also reproduce the markings.
+//
+
+package server
+
+import (
+	"net"
+	"os"
+	"syscall"
+	"testing"
+)
+
+func createTestSocket(t *testing.T, sockPath string) *net.UnixListener {
+	addr := &net.UnixAddr{Name: sockPath, Net: "unixpacket"}
+	sock, err := net.ListenUnix("unixpacket", addr)
+	if err != nil {
+		t.Fatalf("Couldn't set up test socket: %v", err)
+	}
+
+	err = os.Chmod(sockPath, 0777)
+	if err != nil {
+		cleanupTestSocket(sockPath, sock)
+		t.Fatalf("Unable to set permissions on test socket: %v", err)
+	}
+
+	return sock
+}
+
+func cleanupTestSocket(path string, sock *net.UnixListener) {
+	sock.Close()
+	syscall.Unlink(path)
+}
+
+func TestCheckDrpcClientSocketPath_Empty(t *testing.T) {
+	err := checkDrpcClientSocketPath("")
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+}
+
+func TestCheckDrpcClientSocketPath_BadPath(t *testing.T) {
+	err := checkDrpcClientSocketPath("/not/a/real/path")
+
+	if err == nil {
+		t.Fatal("Expected an error, got nil")
+	}
+}
+
+func TestCheckDrpcClientSocketPath_DirNotSocket(t *testing.T) {
+	path := "/tmp/drpc_test.sock"
```

Comment:
ioutil.TempDir() would be a better choice here for two reasons: 1) When the first param is an empty string, it always uses the correct system tempdir rather than assuming /tmp, and 2) It creates a unique directory for the test so that there's no possibility of tests messing with each other.

Reply:
Can do.

Reply:
Fixed